src/server/locals.js

```diff
@@ -0,0 +1,173 @@
+/* globals Map */
+var inspector = require('inspector');
+var async = require('async');
+
+function Locals(config) {
+  if (!(this instanceof Locals)) {
+    return new Locals(config);
+  }
+
+  this.config = config;
+
+  this.initSession();
+}
+
+Locals.prototype.initSession = function() {
+  if (Locals.session) { return; }
+
+  Locals.session = new inspector.Session();
+  Locals.session.connect();
+  Locals.currentErrors = new Map();
+
+  Locals.session.on('Debugger.paused', ({ params }) => {
+    if (params.reason == 'promiseRejection' || params.reason == 'exception') {
+      var key = params.data.description;
+      Locals.currentErrors.set(key, params);
+
+      // Set the max size of the current errors array.
+      // The value should be large enough to preserve each of the errors in
+      // the current cause chain.
+      var CURRENT_ERRORS_MAX_SIZE = 4;
```

Comment:
For now, the codebase is standardized on `var` rather than `let`/`const`, though I think there are a few `const` instances that have been merged in the past. My preference is to stay consistent until the decision is made to move away from `var`.

The value itself shouldn't need to be customized. That's the intent anyway. If there were a demonstrated need (not just for it to be larger, but for it to be larger only in some cases), it could be parameterized then. It's much easier to add to the interface, than to remove or change something.